RUFAS/biophysical/animal/nutrients/nutrition_supply_calculator.py

```diff
@@ -241,6 +265,226 @@ def calculate_actual_maintenance_net_energy(
+        body_weight : float
+            Body weight of a given animal or the average body weight of animals being fed a given ration, kg.
+        total_starch : float
+            Total starch provided in given ration.
+
+        References
+        ----------
+        AN.SUP.2
+        """
+
+        dNDFbase: float = (
+            0.75 * (feed.info.NDF - feed.info.lignin)
+            * (1 - (feed.info.lignin / feed.info.NDF) ** 0.667)) / feed.info.NDF
+
+        dNDF: float = (dNDFbase - 0.0059 * (total_starch - 26) - 1.1
+                       * ((dry_matter_intake / body_weight) - 0.035))
+
+        return dNDF
+
+    @classmethod
+    def calculate_NASEM_dstarch(
+        cls,
+        feed: FeedInRation,
+        dry_matter_intake: float,
+        body_weight: float
+    ) -> float:
+        """
+        NASEM methodology used to calculate starch digestibility (dstarch) of a given feed.
+
+        Adjusts base digestibility.
+
+        Parameters
+        ----------
+        feed : FeedInRation
+            Feed used in ration.
+        dry_matter_intake : float
+            Amount of dry matter intake in given ration, kg.
+        body_weight : float
+            Body weight of a given animal or the average body weight of animals being fed a given ration, kg.
+
+        Returns
+        -------
+        float
+            Digestible starch for a given feed, Mcal/kg.
+
+        References
+        ----------
+        AN.SUP.3
+        """
+        dstarch: float = feed.info.starch_digested * GeneralConstants.PERCENTAGE_TO_FRACTION - 1.0 * (
+            (dry_matter_intake / body_weight) - 0.035)
+
+        return dstarch
+
+    @classmethod
+    def calculate_NASEM_digestible_energy(
+        cls,
+        feeds: list[FeedInRation],
+        dry_matter_intake: float,
+        body_weight: float,
+        total_starch: float
+    ) -> float:
+        """
+        NASEM methodology used to calculate digestible energy for a given ration.
+
+        Parameters
+        ----------
+        feeds : list[FeedInRation]
+            Feeds used in ration.
+        dry_matter_intake : float
+            Amount of dry matter intake in given ration, kg.
+        body_weight : float
+            Body weight of a given animal or the average body weight of animals being fed a given ration, kg.
+        total_starch : float
+            Total starch provided in given ration.
+
+        Returns
+        -------
+        float
+            Digestible energy for a given ration, Mcal/kg.
+
+        References
+        ----------
+        AN.SUP.1, AN.SUP.4, AN.SUP.5, and AN.SUP.6
+        """
+
+        digestible_energy_NASEM_dict: dict[RUFAS_ID, float] = {}
+        dFA: float = 0.73
+        dROM: float = 0.96
+
+        for feed in feeds:
+            if feed.info.Fd_Category is FeedCategorization.NPN_SUPPLEMENT:
+                if feed.info.CP > 0:
+                    NPNsupp: float = feed.info.NPN_source / feed.info.CP
```

Comment:
```suggestion
                    NPN_supply: float = feed.info.NPN_source / feed.info.CP
```
Try to avoid abbreviations.